### PR TITLE
[Enhancement] Refactor ThriftConnectionPool and ThriftRPCRequestExecutor to Unify thrift rpc request

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/system/HeartbeatMgr.java [hunk 1]
```diff
@@ -418,15 +409,13 @@ public BrokerHeartbeatHandler(String brokerName, FsBroker broker, String clientI
...
             try {
-                client = ClientPool.brokerHeartbeatPool.borrowObject(addr);
                 TBrokerPingBrokerRequest request = new TBrokerPingBrokerRequest(TBrokerVersion.VERSION_ONE,
                         clientId);
-                TBrokerOperationStatus status = client.ping(request);
-                ok = true;
+                TBrokerOperationStatus status = ThriftRPCRequestExecutor.call(
+                        ThriftConnectionPool.brokerHeartbeatPool,
+                        new TNetworkAddress(broker.ip, broker.port),
+                        client -> client.ping(request));
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/system/HeartbeatMgr.java:419]: should no retry

File: fe/fe-core/src/main/java/com/starrocks/system/HeartbeatMgr.java [hunk 2]
```diff
@@ -290,7 +284,10 @@ public HeartbeatResponse call() {
                     copiedMasterInfo.setDisabled_disks(((Backend) computeNode).getDisabledDisks());
                     copiedMasterInfo.setDecommissioned_disks(((Backend) computeNode).getDecommissionedDisks());
                 }
-                THeartbeatResult result = client.heartbeat(copiedMasterInfo);
+                THeartbeatResult result = ThriftRPCRequestExecutor.call(
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/system/HeartbeatMgr.java:287]: should no retry

File: fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java [hunk 3]
```diff
@@ -95,23 +93,18 @@ public class BrokerUtil {
...
             TBrokerListPathRequest request = new TBrokerListPathRequest(
                     TBrokerVersion.VERSION_ONE, path, false, brokerDesc.getProperties());
-            TBrokerListResponse tBrokerListResponse = null;
-            try {
-                tBrokerListResponse = client.listPath(request);
-            } catch (TException e) {
-                reopenClient(client);
-                tBrokerListResponse = client.listPath(request);
-            }
+            TBrokerListResponse tBrokerListResponse = ThriftRPCRequestExecutor.callNoRetry(
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java:99]: should retry

File: fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java [hunk 4]
```diff
@@ -184,20 +175,16 @@ public static List<String> parseColumnsFromPath(String filePath, List<String> co
...
+                    ThriftConnectionPool.brokerPool,
+                    address,
+                    client -> client.listPath(request));
+
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java:187]: should retry

File: fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java [hunk 5]
```diff
@@ -216,13 +203,11 @@ public static byte[] readFile(String path, BrokerDesc brokerDesc) throws UserExc
...
             TBrokerOpenReaderRequest tOpenReaderRequest = new TBrokerOpenReaderRequest(
                     TBrokerVersion.VERSION_ONE, path, 0, clientId, brokerDesc.getProperties());
-            TBrokerOpenReaderResponse tOpenReaderResponse = null;
-            try {
-                tOpenReaderResponse = client.openReader(tOpenReaderRequest);
-            } catch (TException e) {
-                reopenClient(client);
-                tOpenReaderResponse = client.openReader(tOpenReaderRequest);
-            }
+            TBrokerOpenReaderResponse tOpenReaderResponse = ThriftRPCRequestExecutor.callNoRetry(
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java:206]: should retry

File: fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java [hunk 6]
```diff
@@ -232,18 +217,15 @@ public static byte[] readFile(String path, BrokerDesc brokerDesc) throws UserExc
...
             TBrokerPReadRequest tPReadRequest = new TBrokerPReadRequest(
                     TBrokerVersion.VERSION_ONE, fd, 0, fileSize);
-            TBrokerReadResponse tReadResponse = null;
-            try {
-                tReadResponse = client.pread(tPReadRequest);
-            } catch (TException e) {
-                reopenClient(client);
-                tReadResponse = client.pread(tPReadRequest);
-            }
+            TBrokerReadResponse tReadResponse = ThriftRPCRequestExecutor.callNoRetry(
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java:220]: should retry

File: fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java [hunk 7]
```diff
@@ -252,32 +234,24 @@ public static byte[] readFile(String path, BrokerDesc brokerDesc) throws UserExc
         } finally {
             // close reader
             if (fd != null) {
-                failed = true;
-                TBrokerCloseReaderRequest tCloseReaderRequest = new TBrokerCloseReaderRequest(
-                        TBrokerVersion.VERSION_ONE, fd);
                 TBrokerOperationStatus tOperationStatus = null;
                 try {
-                    tOperationStatus = client.closeReader(tCloseReaderRequest);
+                    TBrokerCloseReaderRequest tCloseReaderRequest = new TBrokerCloseReaderRequest(
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java:239]: should retry

File: fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java [hunk 8]
```diff
@@ -344,176 +318,109 @@ public static void writeFile(String srcFilePath, String destFilePath,
...
             TBrokerDeletePathRequest tDeletePathRequest = new TBrokerDeletePathRequest(
                     TBrokerVersion.VERSION_ONE, path, brokerDesc.getProperties());
-            TBrokerOperationStatus tOperationStatus = null;
-            try {
-                tOperationStatus = client.deletePath(tDeletePathRequest);
-            } catch (TException e) {
-                reopenClient(client);
-                tOperationStatus = client.deletePath(tDeletePathRequest);
-            }
+            TBrokerOperationStatus tOperationStatus = ThriftRPCRequestExecutor.callNoRetry(
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/common/util/BrokerUtil.java:324]: should retry

**Review by gengjun-git — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/task/AgentBatchTask.java [hunk 9]
```diff
@@ -188,30 +185,24 @@ public void run() {
...
-                // create AgentClient
-                address = new TNetworkAddress(host, port);
-                client = ClientPool.backendPool.borrowObject(address);
                 List<TAgentTaskRequest> agentTaskRequests = new LinkedList<TAgentTaskRequest>();
                 for (AgentTask task : tasks) {
                     agentTaskRequests.add(toAgentTaskRequest(task));
                 }
-                client.submit_tasks(agentTaskRequests);
+
+                ThriftRPCRequestExecutor.call(
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/task/AgentBatchTask.java:193]: no retry?

**Review by gengjun-git — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**